### ISSUE TITLE
Night Light

### ISSUE THREAD

**Issue by Android789515** [pop-os/cosmic-comp#2059]
I'm trying to find a night light that works when using the Cosmic DE.  Redshift does not support Wayland and Gammastep is not working.

Should mention I'm using an Nvidia card.

**Comment by TakkoTheBoss** [comment]
I've already switched to Cosmic as daily driver, but need this feature to be done soon. I know yall are working on it, thanks!!

**Comment by KaKi87** [comment]
I've been waiting for this for so long, then I just now realized my monitors have a night light feature 😂 

Well, it takes 5 button presses on each of my 3 monitors to toggle it on every evening and off every morning, but it's better than nothing.

**Comment by ghost** [comment]
I think this is the one feature preventing me from switching to cosmic right now. I always run a mild red filter.

**Comment by WizardlyMeow** [comment]
Hoping we can see this soon. I saw on the roadmap that this feature isn't going to be considered until the release of Epoch 1? Is that still the case?

**Comment by TakkoTheBoss** [comment]
Any update on this? My corneas have been burning like a vampire in sunlight, and my optometrist is trying to prescribe that I use Windows!!!

**PLEASE** don't make it so I have to use Windows!!!!!! It's depressing!

**Comment by overby-me** [comment]
@TakkoTheBoss consider orange glasses until this is implemented.

**Comment by KaKi87** [comment]
Tried that, doesn't work. But if you have external monitors with a night light feature, that turns out to work well, even though it's a little annoying to toggle.

**Comment by rmarquis** [comment]
The workaround consists in running gammastep manually before launching cosmic with start-cosmic.
It's not ideal because you lose the automatic session startup with greetd, but that actually make cosmic usable at evening and night time.

**Comment by oxysoft** [comment]
ah thats why my eyes are BURNING since installing this shell

**Comment by jjo** [comment]
> The workaround consists in running gammastep manually before launching cosmic with start-cosmic. It's not ideal because you lose the automatic session startup with greetd, but that actually make cosmic usable at evening and night time.

@rmarquis could you please provide more details on how you run it?
asking because you can't actually run `gammastep` before `start-cosmic` per-se (there's no wayland running at that stage), then when the full DE is bootstrapped it seems to be late to run it, i.e. _where_ do you hook `gammastep` invocation?
thanks!

**Comment by nedimzecic** [comment]
Since we will be waiting for more robust color-management for some time I will provide my workaround. I've made small utility here: https://github.com/nedimzecic/gammad which can help users with Intel GPU, probably AMD too. This obviously won't work with Nvidia. The idea is that through DRM you can modify gamma ramp and this should be done before greeter starts. Therefore it has to be injected in /usr/bin/cosmic-greeter-start. There are several downsides of this approach: whenever cosmic-greeter package is updated it will overwrite cosmic-greeter-start script, if you wish to change temperature reboot is required and nvidia is not supported.

![Image](https://github.com/user-attachments/assets/9cf77a34-19ca-42b3-a719-828fb7e945a1)

![Image](https://github.com/user-attachments/assets/ac9a2202-0453-40ee-8934-8753576fa5a3)

**Comment by rmarquis** [comment]
> [@rmarquis](https://github.com/rmarquis) could you please provide more details on how you run it? asking because you can't actually run `gammastep` before `start-cosmic` per-se (there's no wayland running at that stage), then when the full DE is bootstrapped it seems to be late to run it, i.e. _where_ do you hook `gammastep` invocation? thanks!

Sorry, I missed your message. I do not understand how it works, but indeed gammastep works before running start-cosmic. I'm using greetd-agreety (default console based) instead of comsic-greeter, if that helps. This might also be related to some specific hardware setup (nVidia/integrated Intel in my case).

**Comment by SpasMilenkov** [comment]
> probably AMD too

Just writing to confirm that it, in fact, works with AMD, tested on IGPU with ryzen 7 5700U

**Comment by SteamRex25** [comment]
hope we get an nvidia compatible solution soon

**Comment by srid** [comment]
> There are several downsides of this approach: whenever cosmic-greeter package is updated it will overwrite cosmic-greeter-start script, if you wish to change temperature **reboot** is required and nvidia is not supported.

Just wanted to confirm that you can also logout and login back again, instead of rebooting.

**Comment by ericrshields** [comment]
I've used F.lux for decades (on Windows) and they have a linux version. Any chance it works for Cosmic? https://justgetflux.com/linux.html

**Comment by overby-me** [comment]
@ericrshields F.lux will not work, since it only supports X11

**Comment by GuyDebord** [comment]
Night Light is probably the most urgent accessibility feature lacking in Cosmic

**Comment by Alex-Bujorianu** [comment]
It's been over a year with no progress. I'm switching to KDE

**Comment by Merlin-Clos** [comment]
I hope the feature come soon

**Comment by jokokucing** [comment]
following

**Comment by TakkoTheBoss** [comment]
GTA6 will release before we get a Nightlight. 😢

**Comment by leviport** [comment]
We've made it clear that we will work on this feature after Epoch 1. Please stop spamming this thread, as doing so will not change the planned time frames. I will have to lock this thread if it continues to be so noisy.

**Comment by KaKi87** [comment]
It just makes no sense to delay this feature so much, especially since other color accessibility features have been implemented, and considering there are no third-party apps that can work on COSMIC.

**Comment by TakkoTheBoss** [comment]
> We've made it clear that we will work on this feature after Epoch 1. Please stop spamming this thread, as doing so will not change the planned time frames. I will have to lock this thread if it continues to be so noisy.

Its an urgent usability and accessibility feature, and the lack of it creates real friction in adoption. Hiding comments and talking about locking a thread that covers a genuine usability blocker is a fast way to lose community trust.

I joke around in here, but people are commenting because their eyes actually hurt, and because Cosmic is good enough that they want to stay on it. This is not spam. This is user feedback from people who care.

Also, to be clear, nothing in this thread actually states that work on Night Light will begin after Epoch 1. Maybe that information exists somewhere else in a roadmap or another issue, but it is not communicated here. Several people hinted at it, but it is not written in any official way on this thread.

When the only guidance people get is scattered hints across other issues, it makes the situation more confusing. Shushing the rabble by locking the thread or hiding comments does not solve anything, and it will not stop new Night Light issues from being opened again and again.

What would really help is one clear and authoritative statement about the plan and the timeline, even if the answer is simply "not soon and here is why." That level of clarity keeps expectations aligned and avoid this frustration.

Right now my corneas feel like they are starring in the final scene of Raiders of the Lost Ark, so any clarity at all would be appreciated.

-A Cosmic Enjoyer

[Edit: formatting]

**Comment by leviport** [comment]
This is a development issue tracker. Not a blog, and not a chat room for GTA6 jokes. These github repos are our main place of work and collaboration, and the extra noise is a bit of a distraction. Sorry if I'm being a grump about this, but night light requests have been quite noisy and numerous.

If the time frame wasn't clarified in this specific issue, it has been in many other places. At any rate, hopefully the plan is now clear.  I want to assure everyone who is eager for this feature that we are also eager to begin working on this feature. However we are a small team, and we have to prioritize things according to our team's available bandwidth. Thank you all for your enthusiasm about COSMIC DE. We hope you aren't left hanging without a blue light filter for too much longer.

**Comment by TakkoTheBoss** [comment]
> This is a development issue tracker. Not a blog, and not a chat room for GTA6 jokes. These github repos are our main place of work and collaboration, and the extra noise is a bit of a distraction. Sorry if I'm being a grump about this, but night light requests have been quite noisy and numerous.
> 
> If the time frame wasn't clarified in this specific issue, it has been in many other places. At any rate, hopefully the plan is now clear. I want to assure everyone who is eager for this feature that we are also eager to begin working on this feature. However we are a small team, and we have to prioritize things according to our team's available bandwidth. Thank you all for your enthusiasm about COSMIC DE. We hope you aren't left hanging without a blue light filter for too much longer.

I accept this answer @leviport , and appreciate the position you're in. Just know that if I don't see it next November 24th, I'm coming back with the GTA6 references. (This last sentence is intended as a *tongue-in-cheek* joke, as the idea of actually waiting 365 days to do this is absurd)

Thanks for the clarification, and have a great Thanksgiving to everyone on this thread and on the Cosmic DE team!

(please don't hide this!)

**Comment by jokeyrhyme** [comment]
as an outsider (and occasional drive-by contributor), i can see that the team very much welcomes user input and feedback and bug reports, yay! <3

but, it's not polite or friendly or useful or fair to expect more than that: we don't have any authority over how other people spend their time or what they work on, and especially so when we're not helping the team put food on the table in anyway whatsoever 

i'd suggest that we simply thank the team for what they've shared so far, and be patient <3

**Comment by TakkoTheBoss** [comment]
> as an outsider (and occasional drive-by contributor), i can see that the team very much welcomes user input and feedback and bug reports, yay! <3
> 
> but, it's not polite or friendly or useful or fair to expect more than that: we don't have any authority over how other people spend their time or what they work on, and especially so when we're not helping the team put food on the table in anyway whatsoever
> 
> i'd suggest that we simply thank the team for what they've shared so far, and be patient <3

I hear what you are saying @jokeyrhyme , but I also want to push back on the idea that users should feel guilty for giving feedback or for wanting clarity on timelines. Nobody here is trying to dictate how anyone spends their time, and nobody is demanding free labor. People are speaking up because they actually use Cosmic daily, they care about it, and they are running into a real accessibility and usability issue.

Healthy open source communities depend on that feedback. Telling people that their presence or input is only valid if it never becomes persistent or inconvenient creates the exact opposite effect of what a community needs. It makes users feel like they are a burden for caring.

I respect the team. I respect the bandwidth constraints. I respect the priorities. What I do not want is the implication that asking for clarity is somehow crossing a line or disrespecting the nature of open source. It is not. It is what good users do.

So, to be absolutely clear: I am patient. I appreciate the work. I am not entitled to anything. But I am also allowed to advocate for my own eyes and my own user experience without feeling like I have worn out my welcome.

That is all I am trying to do.

**Comment by sfiedler0** [comment]
> hope we get an nvidia compatible solution soon

I am using my nvidia card with the nouveau + nvk / zink driver. That works with the mentioned `gammad`. Good luck with the proprietary / nvidia-open drivers (though that utility which here changes the LUT could also be rewritten to use the CTM (color transformation matrix), which worked last time I checked on nvidia-open)...

**Comment by lorods** [comment]
I guess I'll be taking a look at Cosmic in grayscale for the time being then

**Comment by timkuijsten** [comment]
It might be worth reminding that most monitors can change the colors from cool to warm as well, this stop gap works for me for as long as night light is not implemented in popos.

**Comment by KaKi87** [comment]
> It might be worth reminding that most monitors can change the colors from cool to warm as well

[I already did](https://github.com/pop-os/cosmic-comp/issues/2059#issuecomment-3844402323), as well as mentioning the drawbacks of that workaround.

**Comment by ShamilKaleel** [comment]
we need night light

**Comment by amirhosseinbzg** [comment]
Hello, I have been using Cosmic for 3 days but I may need to switch because It does not matter how cool your other features are my priority is the health of my eyes. (My Eyes already have problems)

**Comment by russjr08** [comment]
I just have a pair of "gamer glasses", which are actually just yellow tinted lenses to emulate the night light effect. Then it doesn't matter what screen I'm looking at, whether that's my TV, my screen, etc - it works for them all.

I'm pretty sure the devs already know how important this function is (given the length and discussions that have already occurred here from the System76 team), bumping the thread and alerting everyone who is subscribed isn't going to speed up the feature's implementation.

Edit: Seems like the original comment and their reply to me was deleted by the author. If you want to show your support for the feature, that is fine - up vote or add any other reaction to the first comment of the thread. 

In general, on GitHub it's considered poor etiquette to comment with a "me too" or similar, as it only adds significant noise to the issue thread. It's one thing if you're posting a workaround, like has been previously posted (or even my "hack" with just using yellow tinted glasses) as that is a meaningful contribution but please don't spam the thread. There are plenty of people who are subscribed so that they can be notified of (meaningful) updates regarding this feature. I don't say this to be rude or of the sorts, but its worthwhile keeping this in mind when commenting.

**Comment by codingbeast** [comment]
> Hello, I have been using Cosmic for 3 days but I may need to switch because It does not matter how cool your other features are my priority is the health of my eyes. (My Eyes already have problems)

its really true, night light ( blue light filter) is more important then other features  as it don't make any sense because this is a core feature . we can't spend more time without blue light filter .

**Comment by leviport** [comment]
As stated numerous times previously, it's coming.

**Comment by seanpk** [comment]
> As stated numerous times previously, it's coming.

I look forward to when it does! In the meantime I'm relying on my main monitor's feature to tune out the blue light.

**Comment by thenry03** [comment]
Just wanted to add I can't wait for this feature. Right now I switch: COSMIC during the day and GNOME at night. But I really really love the way COSMIC feels. However, blue light blocking is a must for me.

Good luck and easy developing!

**Comment by Iugol** [comment]
Hi everyone, I might have found a workaround. If you log into a GNOME on Wayland session, set up Night Light in the display settings as usual, then log out and come back to your COSMIC session, the Night Light filter will still be active. I don't know how, but it works.

**Comment by MrHaroldA** [comment]
FYI: Night Light support is planned for Epoch 3. Every Epoch takes 6-8 months of development, so I'd expect Epoch 3 with night light support somewhere in 2027.

**Comment by KaKi87** [comment]
I still think this should be prioritized. A desktop environment released a stable in 2026 should have this essential feature from the start, not a year later.

**Comment by sfiedler0** [comment]
> Hi everyone, I might have found a workaround. If you log into a GNOME on Wayland session, set up Night Light in the display settings as usual, then log out and come back to your COSMIC session, the Night Light filter will still be active. I don't know how, but it works.

It works because COSMIC (right now) doesn't modify the corresponding color modification tables in the DRM part of the Linux kernel and lets them remain as they were before. 

In my opinion, the better idea is to use [gammad](https://github.com/nedimzecic/gammad), which was linked here at some point in time. You can run this from your framebuffer console (the Linux built-in console, NO display manager; gammad needs to access the DRM which can only be accessed if nothing like a display manager or compositor holds it currently) and set the colors there.  
After that, you can run COSMIC.

**Comment by jjo** [comment]
> > Hi everyone, I might have found a workaround. If you log into a GNOME on Wayland session, set up Night Light in the display settings as usual, then log out and come back to your COSMIC session, the Night Light filter will still be active. I don't know how, but it works.
> 
> It works because COSMIC (right now) doesn't modify the corresponding color modification tables in the DRM part of the Linux kernel and lets them remain as they were before.
> 
> In my opinion, the better idea is to use [gammad](https://github.com/nedimzecic/gammad), which was linked here at some point in time. You can run this from your framebuffer console (the Linux built-in console, NO display manager; gammad needs to access the DRM which can only be accessed if nothing like a display manager or compositor holds it currently) and set the colors there. After that, you can run COSMIC.

As I understand, but please correct me if wrong: the handful of workarounds commented above are "one-off" gamma settings applied before fully starting COSMIC DE, i.e. no timeofday adjustable color temp, correct?

**Comment by rmarquis** [comment]
Looking at gammd, that seems exact. I got the same workaround working with `gammastep`, with the added benefit that you can pass your location data (lat/lon) so the redshift value is computed for the time of the day. It is still not dynamic though.

**Comment by nedimzecic** [comment]
gammad was specifically built for Cosmic to help us all while we are waiting. It works with intel, amd and nvidia (nouveau driver) GPUs basically libdrm. It's injected in cosmic-greeter-start script.

**Comment by MrHaroldA** [comment]
> gammad was specifically built for Cosmic to help us all while we are waiting. It works with intel, amd and nvidia (nouveau driver) GPUs basically libdrm. It's injected in cosmic-greeter-start script.

That has a fixed RGB scale, right? Most of us want it to be dynamic for day/night filtering.

**Comment by shubhamxdev** [comment]
**### Found a Solution** 
this solution is already presented here by @Iugol 
Briefly switch to another TTY, starting a wlroots compositor (I used "WLR_NO_HARDWARE_CURSORS=1 sway --unsupported-gpu"), applying the temperature with "gammastep -O 3500", waiting a few seconds, and then returning to the COSMIC session makes the warm color persist for the entire session. 
No background process is required and COSMIC does not overwrite the DRM gamma LUT afterward. 
This worked on my hybrid Intel/NVIDIA laptop (internal panel driven by Intel)  once sway exited, the system returned to normal. After coming back to normal cosmic session make sure to `pkill sway `.
The only limitation is that the gamma resets after reboot, so it needs to be applied once per session.
Gammd is still better than this but this one came first in my mind .

**Comment by jjo** [comment]
Based on the same principle of the previous comment, I've created (and published built `.deb`) https://github.com/jjo/drm-colortemp.

**TL;DR**: it's a `drm-colortemp-daemon` (run via systemd) that watches `Ctrl-Alt-Fn` console switching, making use of the released DRM lock (from the `cosmic-comp`), so that:
- **Ctrl+Alt+F3** (switches to TTY3) for automatic time-based temperature (config: `/etc/default/drm-colortemp.conf`)
  - Daemon detects the switch and applies gamma (COSMIC releases DRM lock on TTY switch)
  - You immediately press **Ctrl+Alt+F1** (back to COSMIC)
- **Ctrl+Alt+F4** (TTY4) - Force warm/night temperature (e.g. 3500K)
- **Ctrl+Alt+F5** (TTY5) - Force cool/day temperature (6500K)

I.e. no need to log into another VT, _just_ `Ctrl-Alt-Fn` to choose your colortemp ;)

**Comment by TheLevti** [comment]
```
Feb 19 18:01:23 pop-os drm-colortemp-notifier.sh[114995]: Sunset: 18:00 (notify at 00:01)
Feb 19 18:01:23 pop-os drm-colortemp-notifier.sh[114995]: Sunrise: 7:00 (notify at 13:01)
```

what does it mean notify at 00:01? :) Is there some timezone issue?

**Comment by jjo** [comment]
> ```
> Feb 19 18:01:23 pop-os drm-colortemp-notifier.sh[114995]: Sunset: 18:00 (notify at 00:01)
> Feb 19 18:01:23 pop-os drm-colortemp-notifier.sh[114995]: Sunrise: 7:00 (notify at 13:01)
> ```
> 
> what does it mean notify at 00:01? :) Is there some timezone issue?

appreciate if you can create an issue at https://github.com/jjo/drm-colortemp 🙏🏼 , thanks!

**Comment by lorods** [comment]
> > ```
> > Feb 19 18:01:23 pop-os drm-colortemp-notifier.sh[114995]: Sunset: 18:00 (notify at 00:01)
> > Feb 19 18:01:23 pop-os drm-colortemp-notifier.sh[114995]: Sunrise: 7:00 (notify at 13:01)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > what does it mean notify at 00:01? :) Is there some timezone issue?
> 
> appreciate if you can create an issue at https://github.com/jjo/drm-colortemp 🙏🏼 , thanks!

Can confirm the workaround works, thanks! Just passing by to add how one can quickly launch a dummy TTY to apply display temperature:
```sudo openvt --wait --switch -- drm_colortemp -d /dev/dri/cardX -t <colortemp>```

And to avoid typing the full command, one can make it an alias, e. g. on the fish shell:
```alias --save chdrmcolortemp="sudo openvt --wait --switch -- drm_colortemp -d /dev/dri/cardX -t"```

Now it only takes ```chdrmcolortemp K``` to change display temperature.

The only catch seems to be that if I launch KDE Plasma, DRM's gamma gets reset. I can't say this for sure, but when changing TTYs back and forth (from Plasma to a raw one and so on, but not through Cosmic) the textual one has always looked reset to the default, 4500 K probably.

**Comment by tommytz** [comment]
> Based on the same principle of the previous comment, I've created (and published built `.deb`) https://github.com/jjo/drm-colortemp.
> 
> **TL;DR**: it's a `drm-colortemp-daemon` (run via systemd) that watches `Ctrl-Alt-Fn` console switching, making use of the released DRM lock (from the `cosmic-comp`), so that:
> 
>     * **Ctrl+Alt+F3** (switches to TTY3) for automatic time-based temperature (config: `/etc/default/drm-colortemp.conf`)
>       
>       * Daemon detects the switch and applies gamma (COSMIC releases DRM lock on TTY switch)
>       * You immediately press **Ctrl+Alt+F1** (back to COSMIC)
> 
>     * **Ctrl+Alt+F4** (TTY4) - Force warm/night temperature (e.g. 3500K)
> 
>     * **Ctrl+Alt+F5** (TTY5) - Force cool/day temperature (6500K)
> 
> 
> I.e. no need to log into another VT, _just_ `Ctrl-Alt-Fn` to choose your colortemp ;)

Thank you so much for making this! It works really well :)

**Comment by jjo** [comment]
> ```
> Feb 19 18:01:23 pop-os drm-colortemp-notifier.sh[114995]: Sunset: 18:00 (notify at 00:01)
> Feb 19 18:01:23 pop-os drm-colortemp-notifier.sh[114995]: Sunrise: 7:00 (notify at 13:01)
> ```
> 
> what does it mean notify at 00:01? :) Is there some timezone issue?

Released https://github.com/jjo/drm-colortemp/releases/tag/v0.2.7 with the (small, log-only) fix for it.

**Comment by TheLevti** [comment]
> > ```
> > Feb 19 18:01:23 pop-os drm-colortemp-notifier.sh[114995]: Sunset: 18:00 (notify at 00:01)
> > Feb 19 18:01:23 pop-os drm-colortemp-notifier.sh[114995]: Sunrise: 7:00 (notify at 13:01)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > what does it mean notify at 00:01? :) Is there some timezone issue?
> 
> Released https://github.com/jjo/drm-colortemp/releases/tag/v0.2.7 with the (small, log-only) fix for it.

Works now, this is a real life saver!

**Comment by ivanjaros** [comment]
Im just cruising linux distros from windows. Pop(i meant cosmic) is giving me, so far, the least amount of issues. But not having this is a dealbreaker.

**Comment by sfiedler0** [comment]
> Im just cruising linux distros from windows. Pop is giving me, so far, the least amount of issues. But not having this is a dealbreaker.

As said multiple times: It isn't built-in right now, but you can get around it by either using [gammad](https://github.com/nedimzecic/gammad) or [drm-colortemp](https://github.com/jjo/drm-colortemp).

**Comment by KaKi87** [comment]
If it was just a tool to `apt install` and run a command that would be okay, but having to install a whole other desktop environment to log in to first then run the tool then log out then finally log in to COSMIC, that's not an acceaptable workaround, I'm sorry.

**Comment by jjo** [comment]
> If it was just a tool to `apt install` and run a command that would be okay, but having to install a whole other desktop environment to log in to first then run the tool then log out then finally log in to COSMIC, that's not an acceaptable workaround, I'm sorry.

Also still sub-optimal, https://github.com/jjo/drm-colortemp allows you to adjust colortemp by "just" switching to a TTY (Alt-FN) and back (Alt-F1) **without the need to logout/login again** (this is needed for the compositor to release the DRM lock).
There are `.deb` packages ready to install.

**Comment by ivanjaros** [comment]
just pointing the obvious here(also, i'm back on windows)

<img width="1098" height="314" alt="Image" src="https://github.com/user-attachments/assets/4ae38185-452f-451e-9b13-5b159c662144" />

**Comment by KaKi87** [comment]
> Also still sub-optimal, https://github.com/jjo/drm-colortemp allows you to adjust colortemp by "just" switching to a TTY (Alt-FN) and back (Alt-F1)

Yeah, especially since I'd have to go back and forth to check which numeric value I'll like at that point in time 😅

**Comment by jjo** [comment]
> > Also still sub-optimal, https://github.com/jjo/drm-colortemp allows you to adjust colortemp by "just" switching to a TTY (Alt-FN) and back (Alt-F1)
> 
> Yeah, especially since I'd have to go back and forth to check which numeric value I'll like at that point in time 😅

Alt-F3 will always follow sunset setup, ie warm temp from 2000-0800 (defaults), daytime otherwise.

**Comment by heart-rotting** [comment]
@jjo 
Just installed the latest release. Installation was easy and it works great so far. Thanks you!

**Comment by DmitrySkibitsky** [comment]
whole village is waiting, but there are fewer and fewer people

**Comment by rmarquis** [comment]
I wanted a tool that computes the temperature depending on the current location and current time rather than use temperature presets, so I wrote an alternative tool in Zig that does just that: [nebula-drm](https://github.com/rmarquis/nebula-drm).

Like `drm-colortemp` it relies on a systemd service to set gamma table before cosmic starts, and can be refreshed on another TTY to get the correct gamma with a service restart. Unlike it though, it is simpler and allows less manual control (no presets on various TTY) but might be a good alternative if you do not mind restarting the service when needed.

An [AUR package](https://aur.archlinux.org/packages/nebula-drm-git) is available for Arch and derivative distros.

**Comment by ivanjaros** [comment]
@rmarquis yeah, systemd is a spyware now so it's a nonstarter for many from now on.

**Comment by sethCodesSometimes** [comment]
It's spyware? Source? 

Spyware as in the buzzword redditors like, or actual spyware?

**Comment by marco-zan** [comment]
The reddit one. They added a field to store the "user birth date", except that is malicious compliance to the current regulations and nobody is going to fill it or use it. And people are going crazy about it for no reason

**Comment by ivanjaros** [comment]
@marco-zan you sound like the same people that told the world "two weeks to flatten the curve". Anyhow, i do not care as i am back on windows and won't be considering linux again for at least next 5 years.

**Comment by edgar-vincent** [comment]
Please keep this discussion on topic. Many of use subscribe to this issue, so only pertinent information should be added to it.

**Comment by Felipe-TC** [comment]
+1 for this feature. I rely on night light to reduce eye strain in the evening, and it's the only thing I miss from windows. Looking forward to seeing it implemented in COSMIC. Thanks for the great work!

**Comment by a2nath** [comment]
I just saw this issue. Can we implement the color temp change based on time range for now? I understand the feature is complex for dynamic sunset/sunrise trigger based on location. What is exactly gating this issue for the first version?

**Comment by Alex-Bujorianu** [comment]
> I just saw this issue. Can we implement the color temp change based on time range for now? I understand the feature is complex for dynamic sunset/sunrise trigger based on location. What is exactly gating this issue for the first version?

Colour management. However, I'm not sure why the developers care about colour accuracy when, at night light temperatures, it's not possible to reproduce colours accurately, anyway. D50 (5000K black body equivalent) is the ‘yellowest’ that the standards define.

**Comment by The66Monkey** [comment]
Years without Night Light? and the lack of gamma control means no app can provide it. Could you expose basic gamma ramps before full color‑management lands? Every workaround users rely on today is far less accurate than a simple compositor‑level tint, so holding this back feels unnecessarily limiting.

**Comment by metzg** [comment]
There are already filters for colour blindness in cosmic's accessibility settings, would adding similar filters to reduce/remove blue light be an option? I understand that more work is needed for full colour management, and that this would only be a workaround, but this is literally an accessibility feature for me personally (in order to reduce eye strain and headaches, not just at night).

Thanks for all your efforts.

<img width="1037" height="199" alt="Image" src="https://github.com/user-attachments/assets/fa656521-8115-47a3-aae3-baf32bf1cab0" />

**Comment by wueeh** [comment]
Why is this marked low priority, I have eyes to protect. Anyways hold my beer!

**Comment by cetanu** [comment]
> Why is this marked low priority, I have eyes to protect. Anyways hold my beer!

godspeed

**Comment by praisesunsei** [comment]
We need this thing! Thanks for all your team doing!

**Comment by wueeh** [comment]
> Why is this marked low priority, I have eyes to protect. Anyways hold my beer!

I under estimated the depth of this, there has been an attempt; https://github.com/pop-os/cosmic-comp/pull/2417 but it has stalled. With this though I found a solution that works for now. https://github.com/cosmic-nightlight/cosmic-nightlight  kudos to this repository for the good work

**Comment by KaKi87** [comment]
Amazing, thank you for sharing, @danielcwtts you would have been welcome to post here as soon as you released this, so many of us have been waiting for this feature, thank you very much !

**Comment by danielcwtts** [comment]
> Amazing, thank you for sharing, [@danielcwtts](https://github.com/danielcwtts) you would have been welcome to post here as soon as you released this, so many of us have been waiting for this feature, thank you very much !

Thank you for the kind words, and thank you to @wueeh for sharing it! I'm so glad people are finding it useful. I built this project for myself and didn't expect so many people to find it lol. Fingers crossed we'll get a permanent solution soon
